### PR TITLE
checkmarx findings

### PR DESCRIPTION
**Pull request by DivyaDosibhatla**
changed the datatype of passwords from string to char

### CODE REVIEW

File: java/src/main/java/com/cybersource/ws/client/MerchantConfig.java [hunk 1]
```diff
@@ -251,14 +251,15 @@ public MerchantConfig(Properties _props, String _merchantID)
...
-        keyPassword = getProperty(merchantID, "keyPassword");
+        kp = getProperty(merchantID, "keyPassword").toCharArray();
         sendToProduction = getBooleanProperty(merchantID, "sendToProduction", false);
         sendToAkamai = getBooleanProperty(merchantID, "sendToAkamai", false);
         targetAPIVersion = getProperty(merchantID, "targetAPIVersion");
         keyFilename = getProperty(merchantID, "keyFilename");
         serverURL = getProperty(merchantID, "serverURL");
         namespaceURI = getProperty(merchantID, "namespaceURI");
-        password = getProperty(merchantID, "password");
+        if(getProperty(merchantID, "password")!=null)
```
Comment (mahendya1002) [inline java/src/main/java/com/cybersource/ws/client/MerchantConfig.java]: keep curly brackets

File: java/src/main/java/com/cybersource/ws/client/MerchantConfig.java [hunk 2]
```diff
@@ -251,14 +251,15 @@ public MerchantConfig(Properties _props, String _merchantID)
         
         keysDirectory = getProperty(merchantID, "keysDirectory");
         keyAlias = getProperty(merchantID, "keyAlias");
-        keyPassword = getProperty(merchantID, "keyPassword");
+        kp = getProperty(merchantID, "keyPassword").toCharArray();
```
Comment (mahendya1002) [inline java/src/main/java/com/cybersource/ws/client/MerchantConfig.java]: put null check similar to password

File: java/src/main/java/com/cybersource/ws/client/MerchantConfig.java [hunk 3]
```diff
@@ -270,10 +271,11 @@ public MerchantConfig(Properties _props, String _merchantID)
         proxyHost = getProperty(merchantID, "proxyHost");
         proxyPort = getIntegerProperty(merchantID, "proxyPort", DEFAULT_PROXY_PORT);
         proxyUser = getProperty(merchantID, "proxyUser");
-        proxyPassword = getProperty(merchantID, "proxyPassword");
+        if(getProperty(merchantID, "proxyPassword")!=null)
+        pp=getProperty(merchantID, "proxyPassword").toCharArray();
```
Comment (mahendya1002) [inline java/src/main/java/com/cybersource/ws/client/MerchantConfig.java]: keep curly brackets

File: java/src/main/java/com/cybersource/ws/client/MerchantConfig.java [hunk 4]
```diff
@@ -270,10 +271,11 @@ public MerchantConfig(Properties _props, String _merchantID)
...
         enableJdkCert = getBooleanProperty(merchantID, "enableJdkCert", false);
         enableCacert=getBooleanProperty(merchantID, "enableCacert", false);
-        cacertPassword=getProperty(merchantID,"cacertPassword","changeit");
+        cp=getProperty(merchantID,"cacertPassword","changeit").toCharArray();
```
Comment (mahendya1002) [inline java/src/main/java/com/cybersource/ws/client/MerchantConfig.java]: put null check and keep curly brackets.

File: java/src/main/java/com/cybersource/ws/client/MerchantConfig.java [hunk 5]
```diff
@@ -475,7 +477,7 @@ public String getLogString() {
         appendPair(sb, "merchantID", merchantID);
         appendPair(sb, "keysDirectory", keysDirectory);
         appendPair(sb, "keyAlias", keyAlias);
-        appendPair(sb, "keyPassword", keyPassword);
+        appendPair(sb, "keyPassword", new String(kp));
```
Comment (mahendya1002) [inline java/src/main/java/com/cybersource/ws/client/MerchantConfig.java]: can you masked here, similar to proxy password.

File: java/src/main/java/com/cybersource/ws/client/SecurityUtil.java [hunk 6]
```diff
@@ -127,7 +127,7 @@ private static void readAndStoreCertificateAndPrivateKey(MerchantConfig merchant
         }
         
         try {
-            merchantKeyStore.load(new FileInputStream(merchantConfig.getKeyFile()), merchantConfig.getKeyPassword().toCharArray());
+            merchantKeyStore.load(new FileInputStream(merchantConfig.getKeyFile()), merchantConfig.getKeyPassword());
```
Comment (mahendya1002) [inline java/src/main/java/com/cybersource/ws/client/SecurityUtil.java]: I understand that, this is not related to your task but can you please close this FileInputStream.

File: java/src/main/java/com/cybersource/ws/client/SecurityUtil.java [hunk 7]
```diff
@@ -269,7 +269,7 @@ public static void readJdkCert(MerchantConfig merchantConfig, Logger logger)
 		try {
 			FileInputStream is = new FileInputStream(merchantConfig.getKeyFile());
```
Comment (mahendya1002) [inline java/src/main/java/com/cybersource/ws/client/SecurityUtil.java]: close this stream in the finally block.

File: java/src/main/java/com/cybersource/ws/client/SecurityUtil.java [hunk 8]
```diff
@@ -330,7 +330,7 @@ private static void loadJavaKeystore(MerchantConfig merchantConfig, Logger logge
 		try {
 			is = new FileInputStream(merchantConfig.getKeyFile());
```
Comment (mahendya1002) [inline java/src/main/java/com/cybersource/ws/client/SecurityUtil.java]: here as well.

File: java/src/main/java/com/cybersource/ws/client/MerchantConfig.java (hunk 1 above)
Comment (DivyaDosibhatla) [reply java/src/main/java/com/cybersource/ws/client/MerchantConfig.java]: added the brackets

File: java/src/main/java/com/cybersource/ws/client/MerchantConfig.java (hunk 2 above)
Comment (DivyaDosibhatla) [reply java/src/main/java/com/cybersource/ws/client/MerchantConfig.java]: added the null check

File: java/src/main/java/com/cybersource/ws/client/MerchantConfig.java (hunk 3 above)
Comment (DivyaDosibhatla) [reply java/src/main/java/com/cybersource/ws/client/MerchantConfig.java]: added the curly brackets

File: java/src/main/java/com/cybersource/ws/client/SecurityUtil.java (hunk 6 above)
Comment (DivyaDosibhatla) [reply java/src/main/java/com/cybersource/ws/client/SecurityUtil.java]: closed the File input stream

File: java/src/main/java/com/cybersource/ws/client/SecurityUtil.java (hunk 7 above)
Comment (DivyaDosibhatla) [reply java/src/main/java/com/cybersource/ws/client/SecurityUtil.java]: closed the connection

File: java/src/main/java/com/cybersource/ws/client/SecurityUtil.java (hunk 8 above)
Comment (DivyaDosibhatla) [reply java/src/main/java/com/cybersource/ws/client/SecurityUtil.java]: Its already been closed in the finally block.So , I dint make any changes.

**Pull request closed without merging**